src/navigation/navigation-events.js

```diff
@@ -125,29 +125,39 @@ if (isInBrowser) {
     return originalRemoveEventListener.apply(this, arguments);
   };
 
-  window.history.pushState = patchedUpdateState(window.history.pushState);
-  window.history.replaceState = patchedUpdateState(window.history.replaceState);
-
-  function patchedUpdateState(updateState) {
+  window.history.pushState = patchedUpdateState(
+    window.history.pushState,
+    "pushState"
+  );
+  window.history.replaceState = patchedUpdateState(
+    window.history.replaceState,
+    "replaceState"
+  );
+
+  function patchedUpdateState(updateState, methodName) {
     return function () {
       const urlBefore = window.location.href;
       const result = updateState.apply(this, arguments);
       const urlAfter = window.location.href;
 
       if (!urlRerouteOnly || urlBefore !== urlAfter) {
-        urlReroute(createPopStateEvent(window.history.state));
+        urlReroute(createPopStateEvent(window.history.state, methodName));
       }
 
       return result;
     };
   }
 
-  function createPopStateEvent(state) {
+  function createPopStateEvent(state, originalMethodName) {
     // https://github.com/single-spa/single-spa/issues/224 and https://github.com/single-spa/single-spa-angular/issues/49
     // We need a popstate event even though the browser doesn't do one by default when you call replaceState, so that
-    // all the applications can reroute.
+    // all the applications can reroute. We explicitly identify this extraneous event by setting singleSpa=true and
+    // originalMethodName=<pushState|replaceState> on the event instance.
     try {
-      return new PopStateEvent("popstate", { state });
+      const evt = new PopStateEvent("popstate", { state });
+      evt.singleSpa = true;
```

Comment:
IE11 sometimes goes into the `catch`, and we should account for that.

```js
let evt

try {
  evt = new PopStateEvent('popstate', { state })
} catch (err) {
  evt = document.createEvent("PopStateEvent");
  evt.initPopStateEvent("popstate", false, false, state);
}

evt.singleSpa = true;
evt.singleSpaCause = originalMethodName;
return evt
```

Reply:
Ooops I totally meant to do that. Good 👁️

Reply:
It would save a few bytes and avoid duplication of logic to set the extra properties outside of the try/catch, like I showed above. Could you change this to do it that way?